go/mobile/async.go

```diff
@@ -0,0 +1,214 @@
+package iden3mobile
+
+import (
+	"encoding/json"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/iden3/go-iden3-core/db"
+	log "github.com/sirupsen/logrus"
+)
+
+type (
+	Event interface {
+		OnEvent(string, string, string, error) // Type, Ticket ID, Data(json), error
+	}
+
+	Callback interface {
+		VerifierResponse(bool, error)
+		RequestClaimResponse(*Ticket, error)
+	}
+
+	ticketInterface interface {
+		isDone(*Identity) (bool, string, error)
+		marshal() ([]byte, error)
+		unmarshal([]byte) error
+	}
+
+	Ticket struct {
+		Id                   string
+		LastChecked          int64
+		Type                 string
+		handler              ticketInterface
+		HandlerMarshaledData []byte
+	}
+)
+
+func (i *Identity) addTicket(t *Ticket) {
```

Comment:
Since last update, what used to be TicketMap now is replaced by the identity storage. Apart from that, almost all the methods that will be callable from Android will hang on the Identity struct, since this is more convenient on the Java side of things.

I don't mind puting all the methods below the struct where they belong, but is it always how it should be done? (even if a struct has tons of methods?)

Reply:
TODO in #27